README.md

```diff
@@ -2426,19 +2426,82 @@ ari.sounds.list()
 
 ### Events
 
-Event listeners can be registered on the client as well as on resource instances. Note that resource instance listeners are tied to the instance while client listeners receive all events of a given type regardless of which resource the event is for.
+Event listeners can be registered on the client as well as on resource instances.
+
+#### Client Events
+
+Client events are received for all events of a given type regardless of which resource the event is for.
+
+```javascript
+ari.on('StasisStart', function (event, channelInstance) {
+    // will be called for all channels that enter the Stasis application
+});
+```
+
+#### Resource Instance Events
+
+Resource instance events are only received for the given type and resource.
+
+Callbacks:
+
+```javascript
+var channel = ari.Channel();
+channel.on('StasisStart', function (event, channelInstance) {
+    // will only be called when the channel above enters the Stasis application
+});
+
+channel.originate(
+  {endpoint: 'PJSIP/endpoint', app: 'applicationName'},
+  function (err, channelInstance) {}
+);
+```
+
+Promises:
+
+```javascript
+var channel = ari.Channel();
+channel.on('StasisStart', function (event, channelInstance) {
+    // will only be called when the channel above enters the Stasis application
+});
+
+channel.originate({endpoint: 'PJSIP/endpoint', app: 'applicationName'})
+  .then(function (channelInstance) {})
+  .catch(function (err) {});
+```
+
+#### Managing Events
+
+When using events, it is important to remember that the callbacks you provide to handle events cannot be garbage collected until they are unregistered as shown below.
+
+```javascript
+var channel = ari.Channel();
+var handler = function (event, channel) {};
+
+channel.on('ChannelDtmfReceived', handler);
+
+// at some point in your application, remove the event listener to ensure the handler function can be garbage collected
+channel.removeListener('ChannelDtmfReceived', handler);
+// or remove all event listeners for a particular event type
+channel.removeAllListeners('ChannelDtmfReceived');
+```
+
+However, certain resource instance events are unregistered for you after a final event has been received for that resource to minimize possible memory leaks.
```

Comment:
You'll run into a problem here with Channels.

Most channels in a Stasis application will never see the `ChannelDestroyed` event - the subscription to the channel is created when the channel enters `Stasis`, and is removed when the channel leaves `Stasis`. As such, the `ChannelDestroyed` event will never be seen by the library, and the subscriptions will persist. The only way you'll see the event is if someone made an explicit subscription to the channel, subscribed to an Endpoint, or subscribed to everything.

I can see two ways of handling this:
1. Maintain a subscription count yourself in the library to channels. A `StasisStart` or `ChannelCreated` bumps the subscription, as does any explicit subscription that gets made. A `StasisEnd` or `ChannelDestroyed` drops the subscription, as does the destruction of an explicit subscription. When the subscription count is 0, purge any listeners.
2. Have the caller determine how they want the lifetime of their listeners to end - on `StasisEnd` or `ChannelDestroyed`. This pushes the event lifetime management onto the caller - which is more work for them - but means you're out of the bookkeeping business.